go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -0,0 +1,304 @@
+package ovn
+
+import (
+	"fmt"
+	"net"
+	"reflect"
+	"strings"
+	"sync"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	egressfirewallapi "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressfirewall/v1"
+
+	"k8s.io/klog"
+)
+
+const (
+	defaultStartPriority = 2000
+)
+
+type egressFirewall struct {
+	sync.Mutex  //not sure if needed
+	name        string
+	namespace   string
+	egressRules []*egressFirewallRule
+}
+
+type egressFirewallRule struct {
+	id     int
+	access egressfirewallapi.EgressFirewallRuleType
+	ports  []*port
```

Comment:
Any reason in particular for using a slice of pointers?

Reply:
no, changed to slice of ports